hw/ip/otbn/doc/_index.md

```diff
@@ -370,6 +403,24 @@ The stack has a maximum depth of eight and the top of the stack is the current l
 
 ## Design Details
 
+### Random Numbers
+
+OTBN is connected to the [Entropy Distribution Network (EDN)]({{< relref "hw/ip/edn/doc" >}}) which can provide random numbers via the `RND` and `URND` CSRs and WSRs.
+
+`RND` provides bits taken directly from the EDN.
+As an EDN request can take time `RND` is backed by a single-entry cache containing the result of the most recent EDN request.
+A read from `RND` empties this cache.
+A prefetch into the cache, which can be used to hide the EDN latency, is triggered on any write to the `RND_PREFETCH` CSR.
+If the cache is empty when a read from `RND` occurs a EDN request will be made if one isn't in progress already (due to the prefetch).
+OTBN will stall until the request provides bits.
+Both the `RND` CSR and WSR take their bits from the same cache, `RND` CSR reads simply discard the other 192 bits on a read.
+When stalling on an `RND` read OTBN will unstall the cycle after `otbn_core` receives WLEN RND data from the EDN.
```

Comment:
Missing comma after "read". Also, maybe "unstall on the cycle..."? Otherwise it sounds like we're unstalling a cycle (whatever that might mean)